src/loading.jl

```diff
@@ -0,0 +1,86 @@
+"""
+    loadleaf!(dst, src, err)
+
+Copy `src` to `dst` or throw `err` when their sizes are mismatched.
+By default, use `copyto!` when `dst` and `src` are arrays.
+When only `dst` is an array, set every element to `src`.
+Otherwise, just return `dst`.
+"""
+loadleaf!(dst, src, err) = dst
+function loadleaf!(dst::AbstractArray, src, err)
+  dst .= src
+  return dst
+end
+function loadleaf!(dst::AbstractArray, src::AbstractArray, err)
+  (size(dst) == size(src)) || throw(err)
+  copyto!(dst, src)
+end
+
+"""
+    loadmodel!(dst, src)
+
+Copy all the parameters (trainable and non-trainable) from `src` to `dst`.
+
+`loadmodel!` recursively walks the [`Functors.children`](@ref) of `dst` and `src`
+calling `loadleaf!` on any pair of children where [`Functors.isleaf`](@ref) is true.
+It throws an error whenever:
+- `dst` and `src` do not share the same fields (at any level)
+- the sizes of leaf nodes are mismatched between `dst` and `src`
```

Comment:
I wonder if "parameter arrays" is clearer, in case you think Dropout has a non-trainable scalar parameter:
```suggestion
Copy the contents of all parameter arrays (trainable and non-trainable) from `src` to `dst`.

Recursively walks `dst` and `src` together, using Functors.jl, throwing an error if these 
do not share the same field names (at any level). Parameter arrays must be of matching size. 
Non-array elements (such as activation functions) need not match.
An all-zero bias array can be copied to or from absent bias, encoded `bias = false`.
```
I don't think "the sizes of leaf nodes are mismatched" is quite accurate since `size(false)` need not match at present. If we do allow that, maybe it should be explained.

I'd give less weight to the details of Functors.